### PR TITLE
Refactor: Migrate schemas to modular architecture and improve test coverage

### PR DESCRIPTION
**Pull request by rahulkarajgikar**
## Summary
- **Migrate schemas from centralized to modular architecture** - Move schemas from `src/core/config/schemas.ts` to their respective module directories (`llm/`, `mcp/`, `systemPrompt/`, etc.)
- **Add comprehensive test coverage** - Create dedicated schema test files for all modules with detailed validation testing
- **Improve test quality** - Replace generic `toThrow()` assertions with specific Zod error code validation
- **Remove deprecated centralized schemas** - Delete `src/core/config/schemas.ts` after successful migration
- Simplifed system prompt module (removed a lot of unused code + improved test coverage)
- move schema validation into SaikiAgent.ts constructor and delete ConfigManager class

### Schema Migration
- `src/core/schemas/llm.ts` → `src/core/llm/schemas.ts`
- `src/core/schemas/mcp.ts` → `src/core/mcp/schemas.ts`
- **New modular schemas**: `systemPrompt/`, `session/`, `storage/`, `tools/`, `agent/`
- **Remove**: `src/core/config/schemas.ts` (868 lines deleted)
- replaced z.infer with z.output (equivalent, easier to understand)

### Test Coverage Improvements
- **Added comprehensive schema tests** for all modules (2,000+ new test lines)
- **MCP schema tests** - Detailed validation with specific Zod error codes
- **Agent schema tests** - Complete validation for composed schemas
- **LLM schema tests** - Provider-specific validation and error handling
- **SystemPrompt schema tests** - Dynamic/static contributor validation

### Architecture Benefits
- **Clear module boundaries** - Schemas co-located with their implementations
- **Reduced coupling** - Modules can evolve schemas independently  
- **Better maintainability** - Schema changes stay within module scope
- **Type safety** - Improved import paths and module isolation

## Test Results
- **All 667 tests passing** ✅
- **New schema tests**: 180+ comprehensive validation tests
- **Error code validation**: Specific Zod error assertions instead of generic throws
- **Real-world scenarios**: Production config validation tests

## Migration Impact
- **Breaking changes**: None - all public APIs maintained
- **Import updates**: Internal imports updated to new schema locations
- **Backward compatibility**: Full compatibility with existing configurations

🤖 Generated with [Claude Code](https://claude.ai/code)

## Summary by CodeRabbit

* **New Features**
  * Introduced refined schema modules for agent, session, storage, system prompt, and tool configurations with enhanced validation and type safety.
  * Added extensive test suites validating schema correctness, defaults, edge cases, and real-world configurations.

* **Refactor**
  * Streamlined configuration and prompt management to rely on validated, strongly-typed schemas.
  * Enforced fixed prompt generator sources and removed dynamic registration capabilities.
  * Updated import paths across the codebase for consistent schema module usage.
  * Simplified prompt manager initialization by removing legacy loading and registration methods.

* **Bug Fixes**
  * Strengthened validation for critical fields such as API keys and backend connection details to prevent misconfiguration.
  * Delegated API key presence validation fully to schema validation for consistency.

* **Chores**
  * Removed deprecated configuration schema files and legacy system prompt loader logic.
  * Cleaned up exports by removing redundant re-exports and unused modules.

* **Documentation**
  * Improved schema annotations and code documentation for better clarity and maintainability.

### CODE REVIEW

File: src/app/config/cli-overrides.ts [hunk 1]
```diff
@@ -3,8 +3,12 @@
...
  */
 
-import { AgentConfigSchema } from '@core/index.js';
-import type { LLMConfig, ValidatedAgentConfig, AgentConfig } from '@core/index.js';
+import {
+    AgentConfigSchema,
+    type ValidatedAgentConfig,
+    type AgentConfig,
+} from '@core/agent/index.js';
+import type { LLMConfig } from '@core/llm/index.js';
```
[the remark this answers is not in the document]
Comment (rahulkarajgikar) [reply src/app/config/cli-overrides.ts:12]: it does export it though

check index.ts re-exports everything from ./schemas.js

File: src/core/mcp/schemas.test.ts [hunk 2]
```diff
@@ -0,0 +1,707 @@
...
+        describe('Basic Validation', () => {
+            it('should accept valid minimal config', () => {
+                const config: SseServerConfig = {
+                    type: 'sse',
+                    url: 'http://localhost:8080/events',
+                };
+
+                const result = SseServerConfigSchema.parse(config);
+                expect(result.type).toBe('sse');
+                expect((result as any).url).toBe('http://localhost:8080/events');
```
[the remark this answers is not in the document]
Comment (rahulkarajgikar) [reply src/core/mcp/schemas.test.ts]: partially correct!

it works for the stdio/sse ones

but for the mcpserverconfig, we need a typecast

read the code and types more carefully

**Comment by rahulkarajgikar** [comment]
next steps:
- standardize error management
- standardize events
- move agent card stuff to a2a folder, agent-state-manager to agent folder and deprecate config folder
